diff --git a/sites/all/modules/contrib/ctools/views_content/plugins/content_types/views_panes.inc b/sites/all/modules/contrib/ctools/views_content/plugins/content_types/views_panes.inc
index 6d1371a..a1c5af6 100644
--- a/sites/all/modules/contrib/ctools/views_content/plugins/content_types/views_panes.inc
+++ b/sites/all/modules/contrib/ctools/views_content/plugins/content_types/views_panes.inc
@@ -271,6 +271,11 @@ function views_content_views_panes_content_type_render($subtype, $conf, $panel_a
   }
 
   if ($allow['exposed_form'] && !empty($conf['exposed'])) {
+    foreach ($conf['exposed'] as $filter_name => $filter_value) {
+      if(!is_array($filter_value)) {
+        $conf['exposed'][$filter_name] = ctools_context_keyword_substitute($filter_value, array(), $contexts);
+      }
+    }
     $view->set_exposed_input($conf['exposed']);
   }
 
