diff --git a/epsacrop.install b/epsacrop.install
index 7574127..5654458 100644
--- a/epsacrop.install
+++ b/epsacrop.install
@@ -36,7 +36,7 @@ function epsacrop_schema() {
 
 /**
  * Implementation of hook_requirements.
- * 
+ *
  * @access public
  * @param string $phase
  * @return array
@@ -49,6 +49,9 @@ function epsacrop_requirements($phase) {
     // loaded yet.
     include_once dirname(__FILE__) . '/epsacrop.module';
   }
+  if (!function_exists('libraries_get_path')) {
+    include_once drupal_get_path('module', 'libraries') . '/libraries.module';
+  }
   $requirements['epsacrop_jcrop']['title'] = $t('EPSA Crop : Jcrop');
   $requirements['epsacrop_json2']['title'] = $t('EPSA Crop : json2.js');
   if ($version = _epsacrop_jcrop_get_version() || _epsacrop_is_json2_exists(FALSE)) {
@@ -58,7 +61,7 @@ function epsacrop_requirements($phase) {
       $requirements['epsacrop_jcrop']['value'] = 'Jcrop version : ' . $version;
       $requirements['epsacrop_jcrop']['severity'] = REQUIREMENT_OK;
     }
-    
+
     if (_epsacrop_is_json2_exists(FALSE)) {
       $requirements['epsacrop_json2']['value'] = 'JSON file (json2.js) OK';
       $requirements['epsacrop_json2']['severity'] = REQUIREMENT_OK;
