diff --git a/epsacrop.module b/epsacrop.module
index 7795528..3034290 100644
--- a/epsacrop.module
+++ b/epsacrop.module
@@ -155,6 +155,9 @@ function epsacrop_dialog($entity_name, $field_name, $bundle, $fid) {
     $styles = _epsacrop_load_styles_by_instance($entity_name, $field_name, $bundle);
   }
 
+  // Allow other modules to alter list of styles displayed in the popup.
+  drupal_alter('epsacrop_load_styles', $styles, $entity_name, $field_name, $bundle);
+
   foreach ($styles['styles'] as $style_name) {
     $style = _epsacrop_load_style($style_name);
     $effect = _epsacrop_get_effect($style);
